tests/integration/nupic/engine/temporal_memory_compatibility_test.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import json
+import unittest
+import numpy
+
+from nupic.regions.TPRegion import TPRegion
+
+from network_creation_common import createAndRunNetwork
+
+
+
+class TemporalMemoryCompatibilityTest(unittest.TestCase):
+
+  def testTMPyCpp(self):
+    """
+    Test compatibility between C++ and Python TM implementation.
+    """
+    results1 = createAndRunNetwork(TPRegion,
+                                   "bottomUpOut",
+                                   checkpointMidway=False,
+                                   temporalImp="tm_py")
+    results2 = createAndRunNetwork(TPRegion,
+                                   "bottomUpOut",
+                                   checkpointMidway=False,
+                                   temporalImp="py")
```

Comment:
This should be `tm_py` for now. We'll change it to `tm_cpp` once that is supported.